### PR TITLE
Refactor nodegroup creation

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Part of #813 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Review by cPu1 — APPROVED** [review]

File: pkg/actions/create/nodegroup.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+					logger.Info("error waiting for nodes to join the cluster; this command was likely run from outside the cluster's VPC as the API server is not reachable, nodegroup(s) should still be able to join the cluster, underlying error is: %v", err)
+					break
+				} else {
+					return err
+				}
+			}
+		}
+
+		logger.Success("created %d managed nodegroup(s) in cluster %q", len(cfg.ManagedNodeGroups), cfg.Metadata.Name)
+	}
```
Comment (aclevername) [inline pkg/actions/create/nodegroup.go:225]: this is a really long function, can some of the logic get separated out into funcs? The parts where we create a new scope (not sure what the proper term is?) in particular look like good places to move the logic into a descriptive function

File: pkg/actions/create/nodegroup.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+	cmdutils.LogRegionAndVersionInfo(meta)
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+
+	if ok, err := ctl.CanOperate(cfg); !ok {
+		return err
+	}
+
```
Comment (aclevername) [inline pkg/actions/create/nodegroup.go:67]: I've seen these bits of logic all over the place, not suggesting as part of this PR but we should consider trying to move this into some shared step that occurs before a commands logic gets invoked. Something analogous to a [middleware](https://github.com/gorilla/mux#middleware)

File: pkg/actions/create/nodegroup.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/actions/create/nodegroup.go]: done

File: pkg/actions/create/nodegroup.go (hunk 2 above)
Comment (michaelbeaumont) [reply pkg/actions/create/nodegroup.go:67]: yeah for sure, there's a lot of duplication here

**Review by aclevername — APPROVED** [review]
LGTM

**Pull request merged**